app/models/hotel.rb

```diff
@@ -1,13 +1,15 @@
 class Hotel < ApplicationRecord
   belongs_to :city
   has_many :reservations, dependent: :destroy
-  has_many :users, through: :reservations, dependent: :destroy
+  has_many :users, through: :reservations
+  has_many :room_types, through: :reservations
 
-  validates :name, presence: true
-  validates :description, presence: true
-  validates :rating, presence: true
-  validates :image, presence: true
-  validates :city_id, presence: true
+  has_one_attached :images, dependent: :destroy
+  # has_one_attached :images do |attachable|
+  #   attachable.variant :thumb, resize_to_limit: [100, 100]
+  # end
 
-  has_one_attached :image
+  def avatar_url
+    Rails.application.routes.url_helpers.url_for(images)
+  end
```

Comment:
Hi @Tadessejemal1 please remove these lines, they are not relevant for the project to make it work and you are never calling this method within the project, I tested it already 👍🏼 and the image upload works fine without that.